Assets/HoloToolkit-Examples/SpectatorView/Scripts/SpectatorViewNetworkManager.cs

```diff
@@ -0,0 +1,24 @@
+﻿using HoloToolkit.Unity.Preview.SpectatorView;
+using UnityEngine.Networking;
+
+/// <summary>
+/// Example NetworkManager implementation. It simply starts tge server if the HoloLens is running the app.
+/// This component represents the NetworkManager of an entire UNET multilens application.
+/// </summary>
+public class SpectatorViewNetworkManager : NetworkManager
+{
+    /// <summary>
+    /// Is the device a host or a client? (HoloLens or mobile?)
+    /// </summary>
+    private bool isHost;
+
+    // Use this for initialization
+    private void Start()
+    {
+        isHost = FindObjectOfType<PlatformSwitcher>().TargetPlatform == PlatformSwitcher.Platform.Hololens;
```

Comment:
What is this PlatformSwitcher and why can't we either cache a reference to it or make it a singleton?

FindObjectOfType is very expensive.

Reply:
Actually, let's change that to be consistent with the rest of the changes and ask the `SpectatorView` for the `isHost` field. Cached reference of course 👍